voxels/Voxel.h

```diff
@@ -0,0 +1,22 @@
+#include <vector>
+#include "external/Vector/vector.h"
+#include <map>
+#include <string>
+
+class Voxel {
```

Comment:
I don't think we'll need a class representing a single voxel. Individual voxels are contained in a larger voxel chunk. The voxel chunk contains a position / rotation (position for now is fine), an attribute list, and some malloced memory containing the voxel data. The position of an individual voxel depends on the location of the data inside the malloced buffer, so positions don't need to be stored explicitly inside each voxel.

The attribute list should be a list of pairs: the first item is an identifier / string, and the second item is a type enum; we should create an enum with types for each type of data we might want to store in an attribute (for example int32, float, vec4, etc).